terraform/aws/kube/kube.tf

```diff
@@ -0,0 +1,202 @@
+provider "kubernetes" {
+  load_config_file = "false"
+
+  host = var.kube_cluster_endpoint
+
+  cluster_ca_certificate = base64decode(var.kube_cert_data)
+
+  config_context_auth_info = "aws"
+  config_context_cluster   = "kubernetes"
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1alpha1"
+    command     = "aws-iam-authenticator"
+    args        = ["token", "-i", "wandb"]
+  }
+}
+
+##########################################
+# Variables
+##########################################
+
+variable "license" {
+  description = "The license string for your local instance."
+  type        = string
+}
+
+variable "wandb_version" {
+  description = "The version of wandb to deploy."
+  type        = string
+  default     = "0.9.30"
+}
+
+variable "kube_cluster_endpoint" {
+  description = "The endpoint for the kube cluster."
+  type        = string
+}
+
+variable "kube_cert_data" {
+  description = "Certificate authority data for the kube cluster."
+  type        = string
+}
+
+variable "file_storage_bucket_name" {
+  description = "The name of the file storage bucket."
+  type        = string
+}
+
+variable "file_storage_bucket_region" {
+  description = "The region in which the file storage bucket resides."
+  type        = string
+}
+
+variable "file_metadata_queue_name" {
+  description = "The name of the file metadata queue."
+  type        = string
+}
+
+variable "database_endpoint" {
+  description = "The endpoint for the database."
+  type        = string
+}
+
+##########################################
+# Kubernetes
+##########################################
+
+resource "kubernetes_deployment" "wandb" {
+  metadata {
+    name = "wandb"
+    labels = {
+      app = "wandb"
+    }
+  }
+
+  spec {
+    strategy {
+      type = "RollingUpdate"
+    }
+
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "wandb"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "wandb"
+        }
+      }
+
+      spec {
+        container {
+          name              = "wandb"
+          image             = "wandb/local:${var.wandb_version}"
+          image_pull_policy = "Always"
+
+          env {
+            name  = "LICENSE"
+            value = var.license
+          }
+          env {
+            name  = "BUCKET"
+            value = "s3:///${var.file_storage_bucket_name}"
+          }
+          env {
+            name  = "BUCKET_QUEUE"
+            value = "sqs://${var.file_metadata_queue_name}"
+          }
+          env {
+            name  = "AWS_REGION"
+            value = var.file_storage_bucket_region
+          }
+          env {
+            name  = "MYSQL"
+            value = "mysql://wandb:wandb_root_password@${var.database_endpoint}/wandb_local"
```

Comment:
We should make the database password a variable with a default.

Reply:
Done